### PR TITLE
Deprecate mulithreading in bootstrap

### LINKED ISSUE #640: Failures in threaded_replicate test
I am pretty consistently getting about a 50% failure rate on that test.  @palday Do you also get failures on that?  If so, can we suppress it until we find out why it is failing?

### PR DESCRIPTION
**Pull request by palday**
closes #640 

Did behavior change? Did you add need features? If so, please update NEWS.md
- [x] add entry in NEWS.md
- [x] after opening this PR, add a reference and run `docs/NEWS-update.jl` to update the cross-references.

Should we release your changes right away? If so, bump the version:
- [x] I've bumped the version appropriately

### CODE REVIEW

File: src/utilities.jl [hunk 1]
```diff
@@ -124,39 +124,30 @@ end
...
-    If `f()` is not thread-safe or depends on a non thread-safe RNG,
-    then you must set `use_threads=false`. Also note that ordering of replications
-    is not guaranteed when `use_threads=true`, although the replications are not
-    otherwise affected for thread-safe `f()`.
 """
 function replicate(f::Function, n::Integer; use_threads=false, hide_progress=false)
-    # no macro version yet: https://github.com/timholy/ProgressMeter.jl/issues/143
+    use_threads && Base.depwarn(
+        "use_threads is deprecated and will be removed in a future release",
+        :parametricbootstrap,
```
Comment (dmbates) [inline src/utilities.jl:137]: Should this be `:replicate,` ?

**Review by dmbates — APPROVED** [review]
I think that the name of the function needs to be changed in a `@depwarn` call in src/utilities.jl.  Otherwise looks good to go.  Thanks for doing this.

File: src/utilities.jl (hunk 1 above)
Comment (palday) [reply src/utilities.jl:137]: woops, yes, that was a copy-pasto.

Comment (palday) [reply src/utilities.jl:137]:
```suggestion
        :replicate,
```

**Pull request merged**